Source/ending.py

```diff
@@ -0,0 +1,96 @@
+    '''Creates basic, text-only ending'''
+    font50 = pygame.font.Font(None, 50)
+    font20 = pygame.font.Font(None, 20)
+    if victory == 0 or victory == 2:
+        screenScroll(screen, (0,0,0))
+        screen.blit(font50.render("Looks like your portfolio", 1, (255,255,255)), (300, 250))
+        screen.blit(font50.render("could use some improvement", 1, (255,255,255)), (280, 350))
+        Done = False
+        while not Done:
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    Done = True
+            for i in range(len(levels)):
+                screen.blit(font20.render(levels[i], 1, (200,200,200)), (100, 450+i*25))
+            pygame.display.flip()
+    if victory == 1:
+        screenScroll(screen, (255,255,255))
+        Done = False
+        while not Done:
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    Done = True
+            screen.blit(font50.render("Good job!", 1, (0,0,0)), (450, 250))
+            screen.blit(font50.render("You're the ultimate Oliner!'", 1, (0,0,0)), (300, 350))
+            screen.blit(font20.render('All course levels maxed. :)', 1, (0,0,0)), (450, 500))
+            pygame.display.flip()
+            pygame.event.pump()
+
+#not implemented yet
+def graduation():
+    '''Shows graphics for graduating. Used in all endings'''
+    screenScroll(screen, (255,255,255))
+    font100 = pygame.font.Font(None, 100)
+    screen.blit(font100.render("Congratulations,", 1, (0,0,0)), (200, 175))
+    screen.blit(font100.render("Olin Grad!", 1, (0,0,0)), (325, 325))
+    grad_hat = pygame.transform.scale(pygame.image.load('Ending_pics/grad_hat.png'), (250, 200))
+    screen.blit(grad_hat,(350,400))
+    pygame.display.flip()
+    pygame.time.delay(2000)
+    screenScroll(screen, (255,255,255))
+    font50 = pygame.font.Font(None, 50)
+    screen.blit(font100.render("You got your degree...", 1, (0,0,0)), (150, 300))
+    pygame.display.flip()
+    pygame.time.delay(2000)
+    screenScroll(screen, (0,0,0))
+    screen.blit(font100.render("But how will you fair", 1, (255,255,255)), (150, 300))
+    screen.blit(font100.render("in the real world?", 1, (255,255,255)), (170, 400))
+    pygame.display.flip()
+    pygame.time.delay(2000)
+
+
+def badEnding():
+    '''Shows graphics for bad ending. Called when average is below ___.
+    You tried? Desk jobs are all the rage these days.'''
+
+
+
+def middleEnding():
+    '''Shows graphics for middle ending. Called when average is above ___ but levels are not maxed.
+    You are a leading engineer at your own startup'''
+
+
+
+def goodEnding():
+    '''Shows graphics for good ending. Called when levels are all maxed.
+    You are the next Steve Jobs.'''
+
+
+def determine(victory, levels):
+    '''Determines ending.
+    Victory is value 0, 1, or 2. 0 is bad, 2 is perfect.
+    Levels is a list of strings. Print out with for loop'''
```

Comment:
For code that documents itself without as many explicit docstrings and comments, consider:

```python
BAD_OUTCOME = 'bad'
MIDDLE_OUTCOME = 'middle'
PERFECT_OUTCOME = 'perfect'
…
    if victory == MIDDLE_OUTCOME:
       …
````

This has the advantages that it's easy to see the intent each place the value is used, without having to find the nearest comment that describes it; that you don't need to repeat comments in each function that generates or uses these values (admittedly, not that many places in this example); that printing out `victory` is more helpful even if you've forgotten what represents what; and that you're better protected against a certain class of errors, for example using `4` as a victory value.